src/components/Form/ProposalForm.tsx

```diff
@@ -135,61 +132,53 @@ function ProposalForm({
 
         const parsedRateAmountString = parsedRateAmount.toString();
 
-        const cid = await postToIPFS(
-          JSON.stringify({
-            about: values.about,
-            video_url: values.video_url,
-          }),
-        );
+        
+        const proposal = {
+          about: values.about,
+          video_url: values.video_url,
+        }
 
-        // Get platform signature
-        const signature = await getProposalSignature({
-          profileId: Number(user.id),
-          cid,
-          serviceId: Number(service.id),
-        });
+        let tx, cid, proposalResponse;
+
+        cid = await talentLayerClient?.proposal?.upload(proposal);
```

Comment:
two times upload here

Reply:
the first time is actually to generate just the cid so that the delegate call can consume it